### PR TITLE
Added support for build and test on windows

### PR DESCRIPTION
**Pull request by garyjN7**

### CODE REVIEW

File: packages/babel-polyfill/scripts/build-dist.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const polyfillFile = path.join(distPath, "polyfill.js");
+
+try {
+  fs.statSync(distPath);
+}
+catch (e) {
+  fs.mkdirSync(distPath);
+}
+
+const browserifyCommand = `${path.join(thisBase, "lib", "index.js")} \
```
Comment (chrisprice) [inline packages/babel-polyfill/scripts/build-dist.js]: For consistency I'd be tempted to extract this expression to a variable

File: packages/babel-runtime/scripts/build-dist.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ var each       = require("lodash/each");
 var fs         = require("fs");
 var _          = require("lodash");
 
+var join       = require("path").join;
```
Comment (chrisprice) [inline packages/babel-runtime/scripts/build-dist.js:6]: Unless the changes to this file are critical, I'd be tempted to remove them from this PR to simplify it

File: scripts/run-tests.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+"use strict";
+
+require("./babel-register.js");
```
Comment (chrisprice) [inline scripts/run-tests.js:4]: Is this required?

File: scripts/run-tests.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const glob = require("glob");
+
+const testGrep = process.env.npm_config_TEST_GREP;
+const testPackage = process.env.npm_config_TEST_ONLY;
+const testSearch = `packages/${testPackage || "*"}/test/*.js`
+
+const mocha = new Mocha();
+
+function getTestFiles() {
+  glob.sync(testSearch).forEach(function(file) {
```
Comment (chrisprice) [inline scripts/run-tests.js]: Fat arrows?

File: scripts/run-tests.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    mocha.addFile(file);
+  });
+}
+
+function runTests() {
+  testGrep && mocha.grep(testGrep);
```
Comment (chrisprice) [inline scripts/run-tests.js]: Personal preference, I prefer if statements. Leave the `&&` to the obfuscator!

File: scripts/run-tests.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  mocha.ui("tdd").reporter("dot").timeout(10000).run(function(failures) {
+    process.on("exit", function() {
```
Comment (chrisprice) [inline scripts/run-tests.js]: How come you wait for `process.on('exit'`?

**Comment by chrisprice** [comment]
The diff stats look a bit weird to me, you've added more than you've deleted... Did you forget to delete the old shell scripts or are they still required?

**Review by chrisprice — CHANGES_REQUESTED** [review]

File: scripts/run-tests.js (hunk 3 above)
Comment (garyjN7) [reply scripts/run-tests.js:4]: Yes

File: scripts/run-tests.js (hunk 5 above)
Comment (garyjN7) [reply scripts/run-tests.js]: I see short circuit evaluation used all the time and seems quite pedestrian;  i'd hardly call it exotic.

File: scripts/run-tests.js (hunk 4 above)
Comment (garyjN7) [reply scripts/run-tests.js]: ?

**Pull request closed without merging**